javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java

```diff
@@ -770,15 +772,15 @@ private static void createSimpleClientMethods(JavaSettings settings, List<Client
 
         // Generate a non-WithResponse overload with non-required parameters only if one of fluent, include Context,
         // and only required parameters is false.
-        if (!settings.isFluent() || !settings.isContextClientMethodParameter() || !generateClientMethodWithOnlyRequiredParameters) {
+        if (!isSync || !settings.isFluent() || !settings.isContextClientMethodParameter() || !generateClientMethodWithOnlyRequiredParameters) {
```

Comment:
async still need this overload

Reply:
reverted.

note that it would have effect on inner clients of Fluent Premium.

async overload would be
```java
listKeysAsync(String resourceGroupName, String accountName)

listKeysWithResponseAsync(String resourceGroupName, String accountName, ListKeyExpand expand, Context context)
```

but no
```java
listKeysAsync(String resourceGroupName, String accountName, ListKeyExpand expand)
```

Reply:
this would result in some minor adjustment to handwritten code in premium, on next regen.